hdfs/src/main/scala/akka/stream/alpakka/hdfs/model.scala

```diff
@@ -144,6 +153,21 @@ object FilePathGenerator {
   def apply(f: (Long, Long) => String, temp: String = DefaultTempDirectory): FilePathGenerator =
     new FilePathGenerator {
       val tempDirectory: String = temp
+      val newPathForEachFile: Boolean = NewPathForEachFile
+      def apply(rotationCount: Long, timestamp: Long): Path = new Path(f(rotationCount, timestamp))
+    }
+
+  /**
+   * Scala API: creates [[FilePathGenerator]] to rotate output
+   *
+   * @param f    a function that takes rotation count and timestamp to return path of output
+   * @param pathForEachFile turns on/off a possibility to create a new path for each file
+   *                        which depends on rotation count and timestamp supplied by hdfs messages
+   */
+  def apply(f: (Long, Long) => String, pathForEachFile: Boolean): FilePathGenerator =
+    new FilePathGenerator {
+      val tempDirectory: String = DefaultTempDirectory
+      val newPathForEachFile: Boolean = pathForEachFile
```

Comment:
I'm confused about the purpose of `newPathForEachFile`.  If a timestamp extractor perhaps that should imply that the target path is evaluated per record.